akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala

```diff
@@ -345,6 +345,37 @@ object Sink {
       combineRest(2, rest.iterator)
     })
 
+  /**
+   * Combine two sinks with fan-out strategy like `Broadcast` or `Balance` and returns `Sink`.
+   */
+  def combineMat[T, U, M1, M2, M](first: Sink[U, M1], second: Sink[U, M2])(
+      strategy: Int => Graph[UniformFanOutShape[T, U], NotUsed])(combine: (M1, M2) => M): Sink[T, M] =
+    Sink.fromGraph(GraphDSL.createGraph(first, second)(combine) { implicit b => (shape1, shape2) =>
+      import GraphDSL.Implicits._
+      val d = b.add(strategy(2))
+      d.out(0) ~> shape1
+      d.out(1) ~> shape2
+      new SinkShape[T](d.in)
+    })
+
+  /**
+   * Combine several sinks with fan-out strategy like `Broadcast` or `Balance` and returns `Sink`.
+   */
+  def combine[T, U, M](sinks: immutable.Seq[Sink[U, M]])(
+      strategy: Int => Graph[UniformFanOutShape[T, U], NotUsed]): Sink[T, immutable.Seq[M]] =
```

Comment:
Same question here, `sinks.size` is already known by the caller.

It would be nice to validate that the created UniformFanOutShape has the right number of outlets but that might not be possible and I guess the GraphDSL will catch that kind of error.

Reply:
Now it not alian with the old `strategy: Int => ` one .